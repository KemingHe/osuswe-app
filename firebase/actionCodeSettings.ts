import type { ActionCodeSettings } from 'firebase/auth';

import { AUTH_VERIFY_ROUTE } from '@/constants/routeConstants';

// Firebase client-side auth (web app only) action code settings.
const actionCodeSettings: ActionCodeSettings = {
  url: `${window.location.origin}${AUTH_VERIFY_ROUTE}`,
  handleCodeInApp: true,
};

export default actionCodeSettings;
