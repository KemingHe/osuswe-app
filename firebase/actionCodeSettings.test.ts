import actionCodeSettings from '@/firebase/actionCodeSettings';

// Use an arrow function as the second argument for vi.mock()
// instead of the mocked module directly.
vi.mock('@/constants/routeConstants', () => ({
  AUTH_VERIFY_ROUTE: '/auth/verify',
}));

beforeAll(() => {
  Object.defineProperty(window, 'location', {
    value: {
      // 'origin' includes: protocol (http), host (localhost), and port (3000).
      origin: 'http://localhost:3000',
    },
    writable: true,
  });
});

describe('actionCodeSettings', () => {
  it('should have the correct URL', () => {
    expect(actionCodeSettings.url).toBe('http://localhost:3000/auth/verify');
  });

  it('should have handleCodeInApp set to true', () => {
    expect(actionCodeSettings.handleCodeInApp).toBe(true);
  });

  it('should match the snapshot', () => {
    expect(actionCodeSettings).toMatchSnapshot();
  });
});
