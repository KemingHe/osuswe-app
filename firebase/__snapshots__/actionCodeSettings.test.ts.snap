// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`actionCodeSettings > should match the snapshot 1`] = `
{
  "handleCodeInApp": true,
  "url": "http://localhost:3000/auth/verify",
}
`;
